[Seah Yi Xun Ryo] iP

Ryo-Seah opened the pull request:
# Update for ChadGPT

> “Sometimes just being a good and decent person, a good friend to a few, a dependable partner to one, a well focused participant in a thoughtfully chosen career, is as heroic and successful as anyone can truly be.” – John Green ([source](https://www.tiktok.com/))

ChadGPT frees your mind of having to remember things you need to do. It's,

- **text-based**
- *easy to learn*
- FAST **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://www.tiktok.com/).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

**Features:**

- Managing tasks
- Managing deadlines (coming soon)
- Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Ryo-Seah committed:
Complete Level-0

Ryo-Seah committed:
Complete Level-1

Ryo-Seah committed:
Complete Level-2

Ryo-Seah committed:
Complete Level-3

Ryo-Seah committed:
Complete Level-4

Ryo-Seah committed:
Complete A-TextUiTesting

Ryo-Seah committed:
Complete Level-5

Ryo-Seah committed:
Complete Level-6

Ryo-Seah committed:
Complete A-Enums

Ryo-Seah committed:
Complete Level-7

Ryo-Seah committed:
Merge branch 'branch-Level-7'

Ryo-Seah committed:
Complete Level-8

Ryo-Seah committed:
Merge branch 'branch-Level-8'

Ryo-Seah committed:
Complete A-moreOOP

Ryo-Seah committed:
Complete A-Packages

Ryo-Seah committed:
Merge branch 'A-Packages'

Ryo-Seah committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into Branch-A-Gradle

Ryo-Seah committed:
Merge branch 'Branch-A-Gradle'

Ryo-Seah committed:
Remove redundant class files

Ryo-Seah committed:
Complete A-JUnit

Ryo-Seah committed:
Merge branch 'Branch-A-JUnit'

Ryo-Seah committed:
Complete A-JavaDoc

Ryo-Seah committed:
Merge branch 'Branch-A-JavaDoc'

Ryo-Seah committed:
Complete A-CodingStandard

Ryo-Seah committed:
Complete Level-9

Ryo-Seah committed:
Merge branch 'Branch-Level-9'

Ryo-Seah committed:
Fix minor code standards that I missed out.

Ryo-Seah committed:
Complete A-CheckStyle

Ryo-Seah committed:
Merge branch 'branch-A-CheckStyle'

Jayne1010 commented on src/main/java/gpt/Storage.java:
```diff
@@ -0,0 +1,110 @@
+                    String startTime = taskData[3];
+                    String endTime = taskData.length > 4 ? taskData[4] : "";
+
+                    switch(type) {
```
good coding standards implemented

Jayne1010 commented on src/main/java/gpt/Gpt.java:
```diff
@@ -0,0 +1,40 @@
+ */
+public class Gpt {
+    private static final String FILE_PATH = "./gpt/data/GPT.txt"; //for packages, need to change this path to root????
+
+
```
Is there too much of a spacing here?

Jayne1010 commented on src/main/java/gpt/Task.java:
```diff
@@ -0,0 +1,191 @@
+     *
+     * @return The completion status of the task.
+     */
+    public boolean isDone() {
```
good naming of boolean

Jayne1010 reviewed (commented):
Overall, I think there were little to no coding standard violation

Ryo-Seah committed:
Complete Level-10

Ryo-Seah committed:
Merge branch 'branch-Level-10'

Ryo-Seah committed:
Fix Checkstyle

SampsonYe1999 commented:
In general, you have followed the coding standards closely. Examples include not importing the whole package java.io.* if not necessary.  Code is easily readable and very easy to follow. Good work!

Ryo-Seah committed:
Complete A-Assertions
Add assertions for non-null checks in the Storage class Task class and
TaskList class

- Ensure that critical variables in the classes are not null
- Improve development-time checks for potential issues
- Assertions are enabled during development for early error detection

Ryo-Seah committed:
Complete A-CodeQuality
Extract processUnmarkCommand and processMarkCommand methods
{Current situation}
Unmark and mark command processing logic is embedded within the parseCommand method, contributing to its length and complexity.

{Why it needs to change}
Separate the unmark and mark command processing logic into their own methods, processUnmarkCommand and processMarkCommand, respectively, to adhere to the Single Responsibility Principle and improve code organization.

{What is being done about it}
Introduce new methods, processUnmarkCommand and processMarkCommand, to handle the unmark and mark commands. These methods will be responsible for updating the task status and generating the appropriate response strings.

{Why it is done that way}
This change isolates the unmark and mark command logic, making the parseCommand method more focused and promoting code readability. It adheres to best practices for organizing code and enhances maintainability.

{Any other relevant info}
Breaking down the logic into smaller methods improves modularity and readability. This makes the codebase more maintainable and easier to work with.

Ryo-Seah committed:
Merge pull request #2 from Ryo-Seah/branch-A-CodeQuality

Complete A-CodeQuality

Ryo-Seah committed:
Merge branch 'branch-A-Assertions'

Ryo-Seah committed:
Complete C-Statistics

Ryo-Seah committed:
Add Ui picture

Ryo-Seah committed:
Added user guide
